### PR TITLE
1903 Add validation for complex default values in mapping tables on import

### LINKED ISSUE #1903: Accommodate Mapping Tables with complex structures when importing
## Background
Currently, default values can be set only for in flat schemas

## Feature
Accommodate Mapping Tables with complex structures and default values when importing

### PR DESCRIPTION
**Pull request by Zejnilovic**
Fixes #1903 

RN: Added function to allow importing of mapping tables with complex default values with more levels.

### CODE REVIEW

**Comment by Zejnilovic** [comment]
Found out a nice little feature. In SchemaField we stopped adding paths for root fields. Creating a bit of trouble for import/export features.

https://github.com/AbsaOSS/enceladus/issues/1905

This is why there is a need for such a fix not reusing the `getAbsolutePath` which is used in Dataset import/export.

File: menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala [hunk 1]
```diff
@@ -91,9 +91,13 @@ class MappingTableService @Autowired() (mappingTableMongoRepository: MappingTabl
...
+      val fields = schema match {
+        case Some(s) => s.fields.flatMap(f => f.getAllChildrenBasePath :+ f.path).toSet
+        case None => Set.empty[String]
+      }
       item.defaultMappingValue.foldLeft(Validation()) { (accValidations, defaultValue) =>
         accValidations.withErrorIf(
-          schema.exists(s => !s.fields.exists(_.getAbsolutePath == defaultValue.columnName)),
+          !fields.contains(defaultValue.columnName),
           "item.defaultMappingValue",
           s"Cannot fiend field ${defaultValue.columnName} in schema")
```
Comment (benedeki) [inline menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala:102]:
```suggestion
          s"Cannot find field ${defaultValue.columnName} in schema")
```

File: data-model/src/main/scala/za/co/absa/enceladus/model/SchemaField.scala [hunk 2]
```diff
@@ -41,4 +41,9 @@ case class SchemaField
   def getAllChildren: Seq[String] = {
     children.flatMap(child => child.getAllChildren :+ child.getAbsolutePath)
   }
+
+  @JsonIgnore
+  def getAllChildrenBasePath: Seq[String] = {
+    children.flatMap(child => child.getAllChildrenBasePath :+ child.path)
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply data-model/src/main/scala/za/co/absa/enceladus/model/SchemaField.scala:47]: Children are a set. Seems natural

File: menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala [hunk 3]
```diff
@@ -91,9 +91,13 @@ class MappingTableService @Autowired() (mappingTableMongoRepository: MappingTabl
 
   private def validateDefaultValues(item: MappingTable, maybeSchema: Future[Option[Schema]]): Future[Validation] = {
     maybeSchema.map(schema => {
+      val fields = schema match {
```
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala:94]: Is map + getOrElse(Seq.empty) better?

[the remark this answers is not in the document]
Comment (benedeki) [reply menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala:94]: Didn't write it down, seems so. Shorter. 🤷

[the remark this answers is not in the document]
Comment (benedeki) [reply menas/src/main/scala/za/co/absa/enceladus/menas/services/MappingTableService.scala:94]: Agreed to keep, as more readable.

File: data-model/src/main/scala/za/co/absa/enceladus/model/SchemaField.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (benedeki) [reply data-model/src/main/scala/za/co/absa/enceladus/model/SchemaField.scala:47]: Ah `seq` not `set`, and yes then it feels natural.

**Review by dk1844 — APPROVED** [review]
LGTM (just read the code). But the tests look like good evidence of the fix.

**Review by AdrianOlosutean — APPROVED** [review]
Tried locally. Works for validating the column name

**Pull request merged**